### DISCUSSION TITLE
Automatically start at boot - not working

### DISCUSSION THREAD

**Discussion by christensonautohome** [tagyoureit/nodejs-poolController#790] [Q&A] [answered]
I've set up my system, using njsPC and the dashPanel.  However I'm having a terrible time trying to get the programs to autostart on a reboot of my Pi.  I have followed both PM2, Systemd and some other random directions (out of desperation) I found deep on the internet.
What steps am I missing or getting wrong with the PM2 method?  I've followed [these steps](https://github.com/tagyoureit/nodejs-poolController/wiki/Automatically-start-at-boot---PM2-&-Systemd) and confirmed (by running pm2) that both are running with pm2.  But when I reboot the system (using pi 3B+) it doesn't seem to start up either njsPC or the dashPanel.

anyone have any suggestions on the steps I should take to get this working?  what are the usual pitfalls?

**Comment by tagyoureit** [comment]
After you reboot, can you type in the command `pm2 ls` and see what processes it thinks it may have running.  Post a screenshot here.

**Comment by rstrouse** [comment]
Also when you ran `pm2 startup` it should have given you a command in the display.  You need to copy this command and run it in the console.  This adds the pm2 paths to the environment.  If you do not do this it will not be able to start at boot.  You should also be sure you do not have systemd and pm2 attempting to start at boot at the same time.  This will lead to pm2 failing.

**Reply by christensonautohome** [reply]
I've wiped my last card and started over again.  I'll reinstall pm2 tonight and double check i ran the command (I'm nearly 100% I did this earlier but will give it a second try).  Does it matter if the programs (njsPC and njsPC panel) are running when I install and set up pm2?

**Reply by rstrouse** [reply, marked as answer]
You should not have the applications started when you run `pm2 start ecosystem.config.js` this will start the applications and they should be running when you type `pm2 save`.  Also are your running REM?  If you are not then you can remove the reference to REM from the ecosystem.config.js file.

**Reply by christensonautohome** [reply]
so I got it working.  Thank you very much, I must have had njsPC running.  Working perfectly now using PM2 method.
